analysis/import/SourceCodeParser/src/main/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/metricwriters/CSVMetricWriter.kt

```diff
@@ -1,11 +1,16 @@
 package de.maibornwolff.codecharta.importer.sourcecodeparser.metricwriters
 
 import de.maibornwolff.codecharta.importer.sourcecodeparser.metrics.ProjectMetrics
+import de.maibornwolff.codecharta.model.Project
 import java.io.Writer
 
 class CSVMetricWriter(private val writer: Writer) : MetricWriter {
 
-  override fun generate(projectMetrics: ProjectMetrics, allMetrics: Set<String>) {
+  override fun generate(projectMetrics: ProjectMetrics, allMetrics: Set<String>, pipedProject: Project?) {
+    if (pipedProject != null) {
```

Comment:
Use exceptions. Throw an `UnsupportedPipeOutputException` or something like that and catch it, where the function was called and move the error-print there.

Reply:
I unresolved this, because I had another idea; we could throw a UnsupportedOperationException exception. That would be the alternative; either return the project from ScourceCodeParser only and no exception (as implemented), or an Exception that indicates that this feature does in fact not exist and no project as result. Which one do you prefer?

Reply:
@alexhunziker 
Letting the user know that they're doing something, that wont work and just keep the state as it was before trying that forbidden command would be the best. So let's keep it that way